bin/hipcc

```diff
@@ -665,192 +618,58 @@ foreach $arg (@ARGV)
+            $targetsStr = `${ROCM_AGENT_ENUM} -t GPU`;
+            $targetsStr =~ s/\n/,/g;
+            $default_amdgpu_target = 0;
         }
     }
-    # Else try using rocm_agent_enumerator
-    if($default_amdgpu_target eq 1)
-    {
-        $ROCM_AGENT_ENUM = "${ROCM_PATH}/bin/rocm_agent_enumerator";
 
-        my $myAgents = `${ROCM_AGENT_ENUM} -t GPU`;
-        my @agentsLine = split('\n', $myAgents);
+    my @targets = split(',', $targetsStr);
 
-        foreach my $val (@agentsLine) {
-            if($val eq "gfx701") {
-                $target_gfx701 = 1;
-                $default_amdgpu_target = 0;
-            }
-            if($val eq "gfx801") {
-                $target_gfx801 = 1;
-                $default_amdgpu_target = 0;
-            }
-            if($val eq "gfx802") {
-                $target_gfx802 = 1;
-                $default_amdgpu_target = 0;
-            }
-            if($val eq "gfx803") {
-                $target_gfx803 = 1;
-                $default_amdgpu_target = 0;
-            }
-            if($val eq "gfx900") {
-                $target_gfx900 = 1;
-                $default_amdgpu_target = 0;
-            }
-            if($val eq "gfx906") {
-                $target_gfx906 = 1;
-                $default_amdgpu_target = 0;
-            }
-            if($val eq "gfx908") {
-                $target_gfx908 = 1;
-                $default_amdgpu_target = 0;
-            }
-            if($val eq 'gfx1010')
-            {
-                $target_gfx1010 = 1;
-                $default_amdgpu_target = 0;
-            }
-            if($val eq 'gfx1012')
-            {
-                $target_gfx1012 = 1;
-                $default_amdgpu_target = 0;
+    if($HIP_PLATFORM eq "hcc") {
+        $GPU_ARCH_OPT = " --amdgpu-target=";
+    } else {
+        $GPU_ARCH_OPT = " --cuda-gpu-arch=";
+    } 
+
+    foreach my $val (@targets) {
+        my $ignore = 0;
+        if   ($val eq "gfx701")  { $HIPCXXFLAGS .= " -D__HIP_ARCH_GFX701__=1 "; }
+        elsif($val eq "gfx801")  { $HIPCXXFLAGS .= " -D__HIP_ARCH_GFX801__=1 "; }
+        elsif($val eq "gfx802")  { $HIPCXXFLAGS .= " -D__HIP_ARCH_GFX802__=1 "; }
+        elsif($val eq "gfx803")  { $HIPCXXFLAGS .= " -D__HIP_ARCH_GFX803__=1 "; }
+        elsif($val eq "gfx900")  { $HIPCXXFLAGS .= " -D__HIP_ARCH_GFX900__=1 "; }
+        elsif($val eq "gfx906")  { $HIPCXXFLAGS .= " -D__HIP_ARCH_GFX906__=1 "; }
+        elsif($val eq "gfx908")  { $HIPCXXFLAGS .= " -D__HIP_ARCH_GFX908__=1 "; }
+        elsif($val eq "gfx90a")  { $HIPCXXFLAGS .= " -D__HIP_ARCH_GFX90A__=1 "; }
+        elsif($val eq 'gfx1010') { $HIPCXXFLAGS .= " -D__HIP_ARCH_GFX1010__=1 "; }
+        elsif($val eq 'gfx1011') { $HIPCXXFLAGS .= " -D__HIP_ARCH_GFX1011__=1 "; }
+        elsif($val eq 'gfx1012') { $HIPCXXFLAGS .= " -D__HIP_ARCH_GFX1012__=1 "; }
+        elsif($val eq 'gfx000')  { $ignore = 1; } # Ignore gfx000 reported by rocm_agent_enumirator.
```

Comment:
I think for any gfx1xxx we will accept. If the arch is not supported by HCC or HIP-Clang, the compiler will emit error msg any way. This saves us the trouble to manually add each new arch.